app/controllers/signup_controller.rb

```diff
@@ -103,29 +109,36 @@ def verify_email_by_code
   end
 
   def change_signup_email_form
-    @email = unverified_user.email_addresses.first.value
-    render :change_signup_email_form
+    @signing_up_user = get_login_state
   end
 
   def change_signup_email_post
+    @signing_up_user = get_login_state
     handle_with(
       ChangeSignupEmail,
-      user: unverified_user,
+      user: current_user,
       success: lambda {
+        set_login_state(email: @handler_result.outputs.email)
         redirect_to verify_email_by_pin_form_path and return
       },
       failure: lambda {
-        @email = unverified_user.email_addresses.first.value
+        @email = current_user.email_addresses.first.value
         render :change_signup_email_form and return
       }
     )
   end
 
-  def check_your_email; end
+  def check_your_email
+    @signing_up_user = get_login_state
+  end
 
   def signup_done
     security_log(:sign_up_successful, form_name: action_name)
-    redirect_back if current_user.is_tutor_user?
+    if current_user.source_application&.name&.downcase&.include?('tutor')
+      redirect_back
+    else
+      redirect_to(profile_path)
+    end
```

Comment:
I know the code did this already, but feels wrong to send them to the profile page rather than where they were before... for example, if they came here by signing up for highlights, shouldn't they go back to Rex after signup?

Reply:
this is just a small rewrite, removing a somewhat confusing method to the (original code)[https://github.com/openstax/accounts/blob/main/app/controllers/signup_controller.rb#L128]

Reply:
I think this `if` is never true here because there's already a before_action that redirects them away if they are a tutor user: `skip_signup_done_for_tutor_users`

You can probably remove the `if`

Reply:
it would have been nice to do this after the tutor sunset - ha!
fixed